charts/immich/templates/common.yaml

```diff
@@ -1,14 +1,140 @@
 ---
 {{- include "bjw-s.common.loader.init" . }}
 
-{{- define "immich.hardcodedValues" -}}
+{{- define "immich.templatedValues" -}}
 # Set the nameOverride based on the release name if no override has been set
 {{ if not .Values.global.nameOverride }}
 global:
   nameOverride: "{{ .Release.Name }}"
 {{ end }}
+
+persistence:
+  {{- if .Values.immich.configuration }}
+  immich-config:
+    type: configMap
+    name: "{{ .Release.Name }}-immich-config"
+    globalMounts:
+      - path: /config
+        readOnly: false
+  {{- end }}
+
+  library:
+    type: persistentVolumeClaim
+    advancedMounts:
+      server:
+        main:
+          - path: /usr/src/app/upload
+
+  machine-learning-cache:
+    enabled: false
+    type: persistentVolumeClaim
+    advancedMounts:
+      machine-learning:
+        main:
+          - path: /cache
+
+
+serviceMonitor:
+  server:
+    serviceName: server
+    endpoints:
+      - port: metrics-api
+        scheme: http
+      - port: metrics-ms
+        scheme: http
+
+service:
+  server:
+    controller: server
+    primary: true
+    ports:
+      http:
+        primary: true
+        port: 2283
+        protocol: HTTP
+      metrics-api:
+        port: 8081
+        protocol: HTTP
+      metrics-ms:
+        port: 8082
+        protocol: HTTP
+
+  machine-learning:
+    controller: machine-learning
+    ports:
+      http:
+        port: 3003
+        protocol: HTTP
+
+
+rawResources:
+  psql:
+    enabled: false
+    apiVersion: postgresql.cnpg.io/v1
+    kind: Cluster
+    spec:
+      spec:
+        imageCatalogRef:
+          apiGroup: postgresql.cnpg.io
+          kind: ImageCatalog
+          name: '{{ .Release.Name }}-vchord-images'
+          major: 17
+        instances: 1
+        postgresql:
+          shared_preload_libraries:
+            - "vchord.so"
+        managed:
+          roles:
+            - name: immich
+              superuser: true
+              login: true
+        bootstrap:
+          initdb:
+            database: immich
+            owner: immich
+            secret:
+              name: immich-psql
+            postInitSQL:
+              - CREATE EXTENSION IF NOT EXISTS vchord CASCADE;
+        storage:
+          size: 8Gi
+          # storageClass: ''
+
+  # use ImageCatalog since vchord image tag does not have pg major version directly `:17`
+  vchord-images:
+    enabled: '{{ .Values.rawResources.psql.enabled }}'
+    apiVersion: postgresql.cnpg.io/v1
+    kind: ImageCatalog
+    spec:
+      spec:
+        images:
+          - major: 17
+            image: ghcr.io/tensorchord/vchord-postgres:pg17-v0.4.3
```

Comment:
immich provides an "official" DB image with all required extensions, which Is probably a safer bet to use?

Reply:
Probably not [yet]? the image is quite custom and tailored towards compse. it will work best if a psql chart is used, but the operator have much more complicated interaction with the container/image so probably safer to keep everything configured by the operator (CNPG).

Looking at the image, it has custom default configs, and startup command.
https://github.com/immich-app/base-images/blob/2a6af883e858d6064716327bb4b429264761c11d/postgres/Dockerfile